mitmproxy/addons/save.py

```diff
@@ -97,7 +97,7 @@ def maybe_rotate_to_new_file(self) -> None:
         new_log_file = Path(path)
         new_log_file.parent.mkdir(parents=True, exist_ok=True)
 
-        f = new_log_file.open(_mode(ctx.options.save_stream_file))
+        f = new_log_file.open(_mode(ctx.options.save_stream_file), buffering=False)
```

Comment:
I think you missed reverting this:
```suggestion
        f = new_log_file.open(_mode(ctx.options.save_stream_file))
```